### ISSUE TITLE
[SETUP-BUG]

### ISSUE THREAD

**Issue by mkspwr** [mlflow/mlflow#5694]
Thank you for submitting an issue. Please refer to our [issue policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md) for information on what types of issues we address.

**Please fill in this installation issue template to ensure a timely and thorough response.**

### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- Edition	Windows 11 Enterprise
Version	21H2
Installed on	‎6/‎5/‎2021
OS build	22000.613
Experience	Windows Feature Experience Pack 1000.22000.613.0

- **MLflow installed from (source or binary)**: binary
- **MLflow version (run ``mlflow --version``)**: 1.25.1
- **Python version**: 3.10.4
- **Exact command to reproduce**:
with(mlflow_start_run(), {
  mlflow_set_tag("mlflow.runName", "randomForest")
  fit <- train(as.factor(Y_AcceptedOffer) ~ ., data=treatedTrain, method="rf", metric=metric, trControl=control) 
  
  jpeg(filename="1.jpeg")
  print(plot(fit))
  dev.off()
  # Save the plot and log it as an artifact
  mlflow_log_artifact("1.jpeg") 
  accu_train <- get_accuracy(fit, treatedTrain)
  accu_test <- get_accuracy(fit, treatedTest)
  
  message("  accu_train: ", accu_train)
  message("  accu_test: ", accu_test)
  message("  splitPercent: ", splitPercent)
  message("  seed: ", seed)
  message("  device: ", getOption("device"))
  mlflow_log_metric("accu_train", accu_train)
  mlflow_log_metric("accu_test", accu_test)
  mlflow_log_param("splitPercent", splitPercent)
  mlflow_log_param("seed", seed)
  mlflow_log_param("method", "adaptive_cv")
  mlflow_log_param("number", "9")
  
})

### Describe the problem
Provide the exact sequence of commands / steps that you executed before running into the problem.

library(mlflow)
#getting this point is done in another script
treatedTrain <- training
treatedTest <- testing

control <- trainControl(method="adaptive_cv", number=9, verboseIter = TRUE,allowParallel = TRUE)
metric <- "Accuracy"

Sys.setenv(MLFLOW_PYTHON_BIN = "C://python")
Sys.setenv(MLFLOW_BIN="C://python//Scripts//mlflow")

with(mlflow_start_run(), {
  mlflow_set_tag("mlflow.runName", "randomForest")
  fit <- train(as.factor(Y_AcceptedOffer) ~ ., data=treatedTrain, method="rf", metric=metric, trControl=control) 
  
  jpeg(filename="1.jpeg")
  print(plot(fit))
  dev.off()
  # Save the plot and log it as an artifact
  mlflow_log_artifact("1.jpeg") 
  accu_train <- get_accuracy(fit, treatedTrain)
  accu_test <- get_accuracy(fit, treatedTest)
  
  message("  accu_train: ", accu_train)
  message("  accu_test: ", accu_test)
  message("  splitPercent: ", splitPercent)
  message("  seed: ", seed)
  message("  device: ", getOption("device"))
  mlflow_log_metric("accu_train", accu_train)
  mlflow_log_metric("accu_test", accu_test)
  mlflow_log_param("splitPercent", splitPercent)
  mlflow_log_param("seed", seed)
  mlflow_log_param("method", "adaptive_cv")
  mlflow_log_param("number", "9")
  
})

### Other info / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by dbczumar** [comment]
@mkspwr What is the error that you're facing? cc @WeichenXu123

**Comment by mkspwr** [comment]
Sorry..forgot to add it earlier 👍 
Here is the error message: 

**Error in wait_for(function() mlflow_rest("experiments", "list", client = client),  : 
  Operation failed after waiting for 10 seconds**

**Comment by dbczumar** [comment]
> Sorry..forgot to add it earlier 👍 Here is the error message:
> 
> **Error in wait_for(function() mlflow_rest("experiments", "list", client = client), : Operation failed after waiting for 10 seconds**

@mkspwr Is there a more detailed stacktrace?

**Comment by mkspwr** [comment]
Here is what I see when I run the traceback() function:

[[1]]
[1] "stop(\"Operation failed after waiting for \", wait, \" seconds\")"

[[2]]
[1] "wait_for(function() mlflow_rest(\"experiments\", \"list\", client = client), "   
[2] "    getOption(\"mlflow.connect.wait\", 10), getOption(\"mlflow.connect.sleep\", "
[3] "        1))"                                                                     

[[3]]
[1] "mlflow_validate_server(client)"

[[4]]
[1] "mlflow_client()"

[[5]]
[1] "mlflow_start_run()"

[[6]]
 [1] "with(mlflow_start_run(), {"                                            
 [2] "    mlflow_set_tag(\"mlflow.runName\", \"randomForest\")"              
 [3] "    fit <- train(as.factor(Y_AcceptedOffer) ~ ., data = treatedTrain, "
 [4] "        method = \"rf\", metric = metric, trControl = control)"        
 [5] "    jpeg(filename = \"1.jpeg\")"                                       
 [6] "    print(plot(fit))"                                                  
 [7] "    dev.off()"                                                         
 [8] "    mlflow_log_artifact(\"1.jpeg\")"                                   
 [9] "    accu_train <- get_accuracy(fit, treatedTrain)"                     
[10] "    accu_test <- get_accuracy(fit, treatedTest)"                       
[11] "    message(\"  accu_train: \", accu_train)"                           
[12] "    message(\"  accu_test: \", accu_test)"                             
[13] "    message(\"  splitPercent: \", splitPercent)"                       
[14] "    message(\"  seed: \", seed)"                                       
[15] "    message(\"  device: \", getOption(\"device\"))"                    
[16] "    mlflow_log_metric(\"accu_train\", accu_train)"                     
[17] "    mlflow_log_metric(\"accu_test\", accu_test)"                       
[18] "    mlflow_log_param(\"splitPercent\", splitPercent)"                  
[19] "    mlflow_log_param(\"seed\", seed)"                                  
[20] "    mlflow_log_param(\"method\", \"adaptive_cv\")"                     
attr(,"truncated")
[1] TRUE

Please let me know if there is a particular command you'd like me to run to generate the right error message

**Comment by WeichenXu123** [comment]
@mkspwr Would you mind share the dataset used in the reproducing code ?

**Comment by mkspwr** [comment]
yes..its the wine_quality.csv which is available at the following link
mlflow-example/wine-quality.csv at master · mlflow/mlflow-example
(github.com)
<https://github.com/mlflow/mlflow-example/blob/master/wine-quality.csv>

**Comment by WeichenXu123** [comment]
A fixing PR for mlflow-R was merged: https://github.com/mlflow/mlflow/pull/5767
Could you build latest mlflow-R from master and retry ? :)
